internal/app/networkd/pkg/reg/reg.go

```diff
@@ -149,3 +153,67 @@ func toCIDR(family uint8, prefix net.IP, prefixLen int) string {
 
 	return ipNet.String()
 }
+
+// Check implements the Health api and provides visibilty into the state of networkd.
+// Ready signifies the daemon (api) is healthy and ready to serve requests.
+func (r *Registrator) Check(ctx context.Context, in *empty.Empty) (reply *healthapi.HealthCheckResponse, err error) {
+	reply = &healthapi.HealthCheckResponse{
+		Messages: []*healthapi.HealthCheck{
+			{
+				Status: healthapi.HealthCheck_SERVING,
+			},
+		},
+	}
+
+	return reply, nil
+}
+
+// Watch implements the Health api and provides visibilty into the state of networkd.
+// Ready signifies the daemon (api) is healthy and ready to serve requests.
+func (r *Registrator) Watch(in *healthapi.HealthWatchRequest, srv healthapi.Health_WatchServer) (err error) {
+	if in == nil {
+		return errors.New("an input interval is required")
+	}
+
+	var (
+		resp   *healthapi.HealthCheckResponse
+		ticker = time.NewTicker(time.Duration(in.IntervalSeconds) * time.Second)
+	)
+
+	for range ticker.C {
```

Comment:
this loop never finishes, not sure why lines below are commented out as they seem to make perfect sense to me

Reply:
Yeah I missed this :(